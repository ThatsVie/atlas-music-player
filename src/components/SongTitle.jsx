import React from 'react';

const SongTitle = () => {
  return (
    <div>
      <h2 className='mb-2 text-2xl leading-8 font-bold'>Painted in Blue</h2>
      <p className='mb-4'>Soul Canvas</p>
    </div>
  );
};

export default SongTitle;
