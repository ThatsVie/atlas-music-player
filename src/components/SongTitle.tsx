import React from 'react';

interface SongTitleProps {
  title: string;
  artist: string;
}

const SongTitle: React.FC<SongTitleProps> = ({ title, artist }) => (
  <div>
    <h2 className='mb-2 text-2xl leading-8 font-bold'>{title}</h2>
    <p className='mb-4'>{artist}</p>
  </div>
);

export default SongTitle;
