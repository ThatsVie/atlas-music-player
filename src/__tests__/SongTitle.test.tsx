import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import SongTitle from "@/components/SongTitle";

describe("SongTitle Component", () => {
  it("matches snapshot with both title and artist", () => {
    // Arrange
    const title = "Test Song";
    const artist = "Test Artist";

    // Act
    const { container } = render(<SongTitle title={title} artist={artist} />);

    // Assert
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with only title", () => {
    // Arrange
    const title = "Test Song";

    // Act
    const { container } = render(<SongTitle title={title} artist="" />);

    // Assert
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with only artist", () => {
    // Arrange
    const artist = "Test Artist";

    // Act
    const { container } = render(<SongTitle title="" artist={artist} />);

    // Assert
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with empty props", () => {
    // Act
    const { container } = render(<SongTitle title="" artist="" />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
