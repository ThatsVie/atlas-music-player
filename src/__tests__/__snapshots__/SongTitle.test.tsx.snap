// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Component > matches snapshot with both title and artist 1`] = `
<div>
  <div>
    <h2
      class="mb-2 text-2xl leading-8 font-bold"
    >
      Test Song
    </h2>
    <p
      class="mb-4"
    >
      Test Artist
    </p>
  </div>
</div>
`;

exports[`SongTitle Component > matches snapshot with empty props 1`] = `
<div>
  <div>
    <h2
      class="mb-2 text-2xl leading-8 font-bold"
    />
    <p
      class="mb-4"
    />
  </div>
</div>
`;

exports[`SongTitle Component > matches snapshot with only artist 1`] = `
<div>
  <div>
    <h2
      class="mb-2 text-2xl leading-8 font-bold"
    />
    <p
      class="mb-4"
    >
      Test Artist
    </p>
  </div>
</div>
`;

exports[`SongTitle Component > matches snapshot with only title 1`] = `
<div>
  <div>
    <h2
      class="mb-2 text-2xl leading-8 font-bold"
    >
      Test Song
    </h2>
    <p
      class="mb-4"
    />
  </div>
</div>
`;
